### DISCUSSION TITLE
DogeCoin Reduced Amount Transaction

### DISCUSSION THREAD

**Discussion by Sajjad-Khoshdooni** [trustwallet/wallet-core#3457] [Q&A]
I use TrustWalletCore to sign dogecoin transaction and i signed more than 1000 transaction succesfully
recently something bad happens I just created and signed transaction with amount of 1587.304 dogecoin but when it broadcasted
i saw amount is 739.48592923 doge
is it possible beacause of so much of utxos cause transaction big size and its make a mess or my inputs are lower than transaction amount or what is the reason?

this is my code snippet
```async function createDOGEransaction(trx, privateKey){
  const { CoinType, HexCoding, AnySigner, BitcoinScript, BitcoinSigHashType } = await initWasm();

  const lockScript = BitcoinScript.lockScriptForAddress(trx.sender_address, CoinType.dogecoin);


  const utxo_ = [];
  trx.utxo.forEach(element => {
    utxo_.push(
      TW.Bitcoin.Proto.UnspentTransaction.create({
        outPoint: TW.Bitcoin.Proto.OutPoint.create({
          hash: HexCoding.decode(element.hash).reverse(),
          index: element.index,
          sequence: element.sequence 
        }),
        script: lockScript.data(),
        amount: element.amount,
      })
    )
  });
  const input = TW.Bitcoin.Proto.SigningInput.create({
    hashType: BitcoinSigHashType.all.value,
    amount: trx.amount,
    toAddress: trx.receiver_address,
    changeAddress: trx.sender_address,
    byteFee: trx.byte_fee,
    utxo: utxo_, 
    privateKey: [HexCoding.decode(privateKey)],
    coinType: CoinType.dogecoin.value
  });

  const encoded = TW.Bitcoin.Proto.SigningInput.encode(input).finish();
  const outputData = AnySigner.sign(encoded, CoinType.dogecoin);
  const output = TW.Bitcoin.Proto.SigningOutput.decode(outputData);

  return HexCoding.encode(output.encoded).substring(2)
  }

**Comment by satoshiotomakan** [comment]
@Sajjad-Khoshdooni thank you for the issue.
Could you please provide all the inputs without the private key (and public key hashes if you wish)?

**Comment by Sajjad-Khoshdooni** [comment]
> @Sajjad-Khoshdooni thank you for the issue. Could you please provide all the inputs without the private key (and public key hashes if you wish)?



This is the transaction 
https://blockchair.com/dogecoin/transaction/db672c66d63c84a14b0d3571e65c48437feeaddc01deac10b76fac317d45fbe4

**Comment by Malikrehman00107** [comment]
Chek UTXO Value :

**Comment by Malikrehman00107** [comment]
// Calculate the total value of UTXOs
double totalUTXOValue = 0.0;
for (const auto& utxo : utxo_) {
    totalUTXOValue += utxo.amount();
}

// Ensure there are sufficient funds
if (totalUTXOValue < trx.amount + trx.byte_fee) {
    // Handle insufficient funds error
    return "Insufficient funds for the transaction.";
}
